### PR TITLE
KD: add phrase detected notification

### PR DESCRIPTION
**Pull request by iganakov**
Add support for IPC4 Phrase Detected notification to KD module.
Fix KD module stream direction issue. Enable building KD module
on MTL platform.

### CODE REVIEW

File: src/include/ipc4/notification.h [hunk 1]
```diff
@@ -81,3 +81,38 @@ union ipc4_notification_header {
...
+		uint32_t dat;
+
+		struct {
+			/**< Final speaker verification score in range of 0..8192 */
+			uint32_t sv_score : 16;
+			uint32_t rsvd1 : 14;
+			uint32_t _reserved_2 : 2;
+		} r;
+	} extension;
+} __packed __aligned(4);
```
Comment (lgirdwood) [inline src/include/ipc4/notification.h]: newline

File: app/boards/intel_adsp_ace15_mtpm.conf [hunk 2]
```diff
@@ -46,6 +46,7 @@ CONFIG_LOG_MODE_DEFERRED=y
 CONFIG_LOG_FUNC_NAME_PREFIX_INF=y
 CONFIG_COMP_VOLUME_WINDOWS_FADE=y
 CONFIG_COMP_UP_DOWN_MIXER=y
+CONFIG_SAMPLE_KEYPHRASE=y
```
Comment (lgirdwood) [inline app/boards/intel_adsp_ace15_mtpm.conf:49]: @kv2019i are samples going in an overlay ?

Comment (kv2019i) [reply app/boards/intel_adsp_ace15_mtpm.conf:49]: @lgirdwood I think we can keep them in board files for now. We currently don't have a separate samples overlay, and the extra overlays don't in general get CI coverage, so better to have this in one of the CI-coverated configurations.

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
CI passed.

**Pull request merged**